### PR TITLE
[flink]support table debezium json format

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

support table debezium json format

### Tests

KafkaDebeziumSyncTableActionITCase

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/utils/JsonSerdeUtil.java [hunk 1]
```diff
@@ -219,6 +221,43 @@ public static <T> JsonNode toTree(T value) {
...
+     * @param pkNames A list of primary key names to be added to the JSON string.
+     * @param pkNamesKey The key under which the primary key names will be added.
+     * @return The JSON string with the added primary key names. If the JSON string is not a valid
+     *     JSON object, or if the list of primary key names is empty or null, the original JSON
+     *     string will be returned.
+     * @throws RuntimeException If an error occurs while parsing the JSON string or adding the
+     *     primary key names.
+     */
+    public static String addPrimaryKeysToJson(
+            String value, List<String> pkNames, String pkNamesKey) {
```
Comment (yuzelin) [inline paimon-core/src/main/java/org/apache/paimon/utils/JsonSerdeUtil.java:237]: We can make this method more common. How about `String putArrayToJsonString(String origin, String key, List<String> values)`

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public boolean isEndOfStream(String nextElement) {
+        return false;
+    }
+
+    @Override
+    public TypeInformation<String> getProducedType() {
+        return TypeInformation.of(new TypeHint<String>() {});
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java:60]: `return BasicTypeInfo.STRING_TYPE_INFO;`

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * This class is used to deserialize byte[] messages into String format, and then add primary key
+ * fields to the JSON string.
+ */
+public class JsonPrimaryKeyDeserializationSchema implements DeserializationSchema<String> {
+
+    public static final String PRIMARY_KEY_NAMES = "pkNames";
+    private final List<String> primaryKeyNames;
+
+    public JsonPrimaryKeyDeserializationSchema(List<String> primaryKeyNames) {
+        this.primaryKeyNames = primaryKeyNames;
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java:40]: Check that `primaryKeyNames` is not empty.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaDebeziumSyncTableActionITCase.java [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.testutils.assertj.AssertionUtils.anyCauseMatches;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** IT cases for {@link KafkaDebeziumSyncTableActionITCase}. */
+public class KafkaDebeziumSyncTableActionITCase extends KafkaActionITCaseBase {
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaDebeziumSyncTableActionITCase.java:38]: We can only test dbz format parsing, so just keep `testSchemaEvolution` and `testComputedColumn` and delete other tests.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java (hunk 2 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java:60]: done.

File: paimon-core/src/main/java/org/apache/paimon/utils/JsonSerdeUtil.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/utils/JsonSerdeUtil.java:237]: done.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaDebeziumSyncTableActionITCase.java (hunk 4 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaDebeziumSyncTableActionITCase.java:38]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ * This class is used to deserialize byte[] messages into String format, and then add primary key
+ * fields to the JSON string.
+ */
+public class JsonPrimaryKeyDeserializationSchema implements DeserializationSchema<String> {
+
+    public static final String PRIMARY_KEY_NAMES = "pkNames";
+    private final List<String> primaryKeyNames;
+
+    public JsonPrimaryKeyDeserializationSchema(List<String> primaryKeyNames) {
+        checkNotNull(primaryKeyNames);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java:42]: It should be not null and not empty?

Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/JsonPrimaryKeyDeserializationSchema.java]: done.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**